test/libsolutil/LP.cpp

```diff
@@ -0,0 +1,314 @@
+	feasible({{"x", 100}});
+
+	addLEConstraint(constant(5), variable("x"));
+	feasible({{"x", 100}});
+
+	addLowerBound("x", 20);
+	feasible({{"x", 100}});
+	addLowerBound("x", 25);
+	feasible({{"x", 100}});
+
+	addUpperBound("x", 20);
+	infeasible();
+}
+
+BOOST_AUTO_TEST_CASE(bounds2)
+{
+	addLowerBound("x", 200);
+	addUpperBound("x", 250);
+	addLowerBound("y", 2);
+	addUpperBound("y", 3);
+	feasible({{"x", 250}, {"y", 3}});
+
+	addLEConstraint(variable("y"), variable("x"));
+	feasible({{"x", 250}, {"y", 3}});
+
+	addEQConstraint(variable("y") + constant(231), variable("x"));
+	feasible({{"x", 234}, {"y", 3}});
+
+	addEQConstraint(variable("y") + constant(10), variable("x") - variable("z"));
+	feasible({{"x", 234}, {"y", 3}});
+
+	addEQConstraint(variable("z") + variable("x"), constant(2));
+	infeasible();
+}
+
+BOOST_AUTO_TEST_CASE(lower_bound)
+{
+	addLEConstraint(constant(1), variable("y"));
+	addLEConstraint(variable("x"), constant(10));
+	addLEConstraint(2 * variable("x") + variable("y"), 2);
+	feasible({{"x", 0}, {"y", 2}});
+}
+
+BOOST_AUTO_TEST_CASE(check_infeasible)
+{
+	addLEConstraint(variable("x"), 3);
+	addLEConstraint(constant(5), variable("x"));
+	infeasible();
+}
+
+BOOST_AUTO_TEST_CASE(unbounded1)
+{
+	addLEConstraint(constant(2), variable("x"));
+	feasible({{"x", 2}});
+}
+
+BOOST_AUTO_TEST_CASE(unbounded2)
+{
+	auto x = variable("x");
+	auto y = variable("y");
+	addLEConstraint(constant(2), x + y);
+	addLEConstraint(x, 10);
+	feasible({{"x", 10}, {"y", 0}});
+}
+
+BOOST_AUTO_TEST_CASE(unbounded3)
+{
+	addLEConstraint(constant(0) - variable("x") - variable("y"), constant(10));
+	feasible({{"x", 0}, {"y", 0}});
+
+	addLEConstraint(constant(0) - variable("x"), constant(10));
+	feasible({{"x", 0}, {"y", 0}});
+
+	addEQConstraint(variable("y") + constant(3), variable("x"));
+	feasible({{"x", 3}, {"y", 0}});
+
+	addLEConstraint(variable("y") + variable("x"), constant(2));
+	infeasible();
+}
+
+
+BOOST_AUTO_TEST_CASE(equal)
+{
+	auto x = variable("x");
+	auto y = variable("y");
+	addEQConstraint(x, y + constant(10));
+	addLEConstraint(x, 20);
+	feasible({{"x", 20}, {"y", 10}});
+}
+
+
+BOOST_AUTO_TEST_CASE(equal_constant)
+{
+	auto x = variable("x");
+	auto y = variable("y");
+	addLEConstraint(x, y);
+	addEQConstraint(y, constant(5));
+	feasible({{"x", 5}, {"y", 5}});
+}
+
+BOOST_AUTO_TEST_CASE(linear_dependent)
+{
+	auto x = variable("x");
+	auto y = variable("y");
+	auto z = variable("z");
+	addLEConstraint(x, 5);
+	addLEConstraint(2 * y, 10);
+	addLEConstraint(3 * z, 15);
+	// Here, they should be split into three independent problems.
+	feasible({{"x", 5}, {"y", 5}, {"z", 5}});
+
+	addLEConstraint((x + y) + z, 100);
+	feasible({{"x", 5}, {"y", 5}, {"z", 5}});
+
+	addLEConstraint((x + y) + z, 2);
+	feasible({{"x", 2}, {"y", 0}, {"z", 0}});
+
+	addLEConstraint(constant(2), (x + y) + z);
+	feasible({{"x", 2}, {"y", 0}, {"z", 0}});
+
+	addEQConstraint(constant(2), (x + y) + z);
+	feasible({{"x", 2}, {"y", 0}, {"z", 0}});
+}
+
```

Comment:
```suggestion

BOOST_AUTO_TEST_CASE(fuzzer)
{
        auto x = variable("x");
        auto y = variable("y");
        addEQConstraint(6 * x + 6 * y, constant(-8));
        sat();
}
```

Reply:
I added `all_equality` which is the same test but with correct expectation (after the fix)